pkg/render/monitor/monitor.go

```diff
@@ -45,35 +48,74 @@ const (
 	CalicoNodePrometheus        = "calico-node-prometheus"
 	ElasticsearchMetrics        = "elasticsearch-metrics"
 	FluentdMetrics              = "fluentd-metrics"
+	TigeraPrometheusObjectName  = "tigera-prometheus"
+	TigeraPrometheusSAName      = "prometheus"
 	TigeraPrometheusDPRate      = "tigera-prometheus-dp-rate"
 	TigeraPrometheusRole        = "tigera-prometheus-role"
 	TigeraPrometheusRoleBinding = "tigera-prometheus-role-binding"
 
-	PrometheusHTTPAPIServiceName = "prometheus-http-api"
-	PrometheusDefaultPort        = 9090
+	PrometheusHTTPAPIServiceName    = "prometheus-http-api"
+	PrometheusDefaultPort           = 9090
+	PrometheusProxyPort             = 9095
+	PrometheusTLSSecretName         = "calico-node-prometheus-tls"
+	calicoNodePrometheusServiceName = "calico-node-prometheus"
+
+	tigeraPrometheusServiceHealthEndpoint = "/health"
 
 	AlertmanagerConfigSecret = "alertmanager-calico-node-alertmanager"
 
 	prometheusServiceAccountName = "prometheus"
 )
 
-func Monitor(cfg *Config) render.Component {
-	return &monitorComponent{
-		cfg: cfg,
+func Monitor(cfg *Config) (render.Component, error) {
+	var err error
+	var tlsSecrets []*corev1.Secret
+	var tlsHash string
+	if cfg.Installation.CertificateManagement == nil {
```

Comment:
I don't think this is supposed to be in the render package, I think this kind of verification is supposed to be done before render.

Are we doing this in other places, or is this new?